src/pseudopeople/entity_types.py

```diff
@@ -10,15 +12,20 @@
 from pseudopeople.configuration import Keys
 from pseudopeople.utilities import ensure_dtype, get_index_to_noise
 
+if TYPE_CHECKING:
+    from pseudopeople.configuration.noise_configuration import NoiseConfiguration
+
 
 def _noise_function_not_implemented(*_args: Any, **_kwargs: Any) -> None:
     pass
 
 
 def default_noise_level_getter(
-    _dataset: "Dataset", configuration: LayeredConfigTree
+    noise_type: str, _dataset: Dataset, configuration: NoiseConfiguration
```

Comment:
Nit: since we're changing this, I would prefer a parameter order of `configuration, dataset, noise_type`.

Also, `_dataset` should be changed to `dataset` since the argument is actually being used in the function now.